zebra-state/src/service/finalized_state/disk_db.rs

```diff
@@ -514,6 +491,43 @@ impl DiskDb {
             .map(|(key, value)| (K::from_bytes(key), V::from_bytes(value)))
     }
 
+    /// Returns the RocksDB ReadOptions with a lower and upper bound for a range.
+    fn zs_iter_opts<R>(range: &R) -> ReadOptions
+    where
+        R: RangeBounds<Vec<u8>>,
+    {
+        use std::ops::Bound::*;
+
+        let mut opts = ReadOptions::default();
+
+        if let Included(bound) | Excluded(bound) = range.start_bound() {
+            opts.set_iterate_lower_bound(bound.clone());
+        };
+
+        match range.end_bound().cloned() {
+            Included(mut bound) => {
+                // Increment the last byte in the vector that is not u8::MAX, or
+                // skip adding an upper bound if every byte is u8::MAX
+                if let Some(increment_idx) = bound.iter().rposition(|&v| v != u8::MAX) {
+                    let increment_byte = bound
+                        .get_mut(increment_idx)
+                        .expect("index should be in bounds");
+                    *increment_byte = increment_byte
+                        .checked_add(1)
+                        .expect("adding 1 should succeed");
```

Comment:
This should be covered by `all_upgrades_and_wrong_commitments_with_fake_activation_heights` using `zs_next_key_value_from()`.

Reply:
But unfortunately that test does not have sufficient coverage for the goals of the ticket.